addons/website_sale/tests/test_website_sale_cart_recovery.py

```diff
@@ -11,3 +11,90 @@ class TestWebsiteSaleCartRecovery(HttpCase):
     def test_01_shop_cart_recovery_tour(self):
         """The goal of this test is to make sure cart recovery works."""
         self.browser_js("/", "odoo.__DEBUG__.services['web_tour.tour'].run('shop_cart_recovery')", "odoo.__DEBUG__.services['web_tour.tour'].tours.shop_cart_recovery.ready", login="portal")
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleCartRecoveryServer(TransactionCase):
+
+    def setUp(self):
+        res = super(TestWebsiteSaleCartRecoveryServer, self).setUp()
+
+        self.customer = self.env['res.partner'].create({
+            'name': 'a',
+            'email': 'a@example.com',
+        })
+        self.recovery_template_default = self.env.ref('website_sale.mail_template_sale_cart_recovery')
+        self.recovery_template_custom1 = self.recovery_template_default.copy()
+        self.recovery_template_custom2 = self.recovery_template_default.copy()
+
+        self.website0 = self.env['website'].create({
+            'name': 'web0',
+            'cart_recovery_mail_template_id': self.recovery_template_default.id,
+        })
+        self.website1 = self.env['website'].create({
+            'name': 'web1',
+            'cart_recovery_mail_template_id': self.recovery_template_custom1.id,
+        })
+        self.website2 = self.env['website'].create({
+            'name': 'web2',
+            'cart_recovery_mail_template_id': self.recovery_template_custom2.id,
+        })
+        self.so0 = self.env['sale.order'].create({
+            'partner_id': self.customer.id,
+            'website_id': self.website0.id,
+            'is_abandoned_cart': True,
+            'cart_recovery_email_sent': False,
+        })
+        self.so1 = self.env['sale.order'].create({
+            'partner_id': self.customer.id,
+            'website_id': self.website1.id,
+            'is_abandoned_cart': True,
+            'cart_recovery_email_sent': False,
+        })
+        self.so2 = self.env['sale.order'].create({
+            'partner_id': self.customer.id,
+            'website_id': self.website2.id,
+            'is_abandoned_cart': True,
+            'cart_recovery_email_sent': False,
+        })
+
+        return res
+
+    def test_cart_recovery_mail_template(self):
+        """Make sure that we get the correct cart recovery templates to send."""
+        self.assertEqual(self.so1._get_cart_recovery_template(),
+                         self.so1.website_id.cart_recovery_mail_template_id,
+                         "We do not return the correct mail template")
+
+        self.assertEqual(self.so2._get_cart_recovery_template(),
+                         self.so2.website_id.cart_recovery_mail_template_id,
+                         "We do not return the correct mail template")
+
+        # Orders that belong to different websites; we should get the default template
+        self.assertEqual((self.so1 | self.so2)._get_cart_recovery_template(),
```

Comment:
It probably doesn't matter much here but `+` is likely a bit more efficient (and readable) than `|` when you know for sure you want the 2 records on the set (same below).